### ISSUE TITLE
switch to a much more vanilla flux solution

### ISSUE THREAD

**Issue by vesparny** [vesparny/morpheus#28]
Morpheus is highly coupled to fluxible.
I really enjoy it, but I think a much more vanilla solution would be the best solution.

In order to get rid of fluxible, we will need to face some problems:
- replace the routing fluxible solution with react-router
- replace fluxible with some other flux lib
- restore the WebApiUtils. look [here](https://github.com/vesparny/morpheus/blob/815f79e76fbe6207e0561a41131b42dc15a6d28c/client/utils/api.js) 

@acdlite any thoughts ?

**Comment by acdlite** [comment]
Obviously I'm biased, but I think Flummox hits all your requirements perfectly. One nice thing about Flummox which may not be noticeable at first is that since it's self-contained, you can replace your old Flux stores and actions one at a time — in other words, have both Fluxible and Flummox stores running side by side during development until they're all replaced. (Does that make sense?)

Also, in order to get feature parity with Fluxible and alt, I need to implement dehydration/rehydration. I've opened up a new issue to track that on the Flummox repo.

https://github.com/acdlite/flummox/issues/5

**Comment by vesparny** [comment]
I'll try to work on it thanks.
Do you think it will work well with immutable.js?
Was wondering about integrating it for stores handling.

**Comment by acdlite** [comment]
Yeah, absolutely. The state API for Stores is designed to be very similar to the API for React components. Here's an working example I made just this past weekend on an internal project for work: https://github.com/parisleaf/parisleaf.com/blob/develop/src/shared/stores/PageStore.js

Eventually, you could even create a new class ImmutableStore that serves as the base for all your stores. I think there's lot of potential for this pattern (+mixins) as a "plugin" strategy.

**Comment by vesparny** [comment]
Thanks, I'll take a look.

I just came through this: https://github.com/goatslacker/iso/blob/master/examples/react-router-flux/server.js
It's an Alt side project and I think it covers what you did by making flummox singleton free.
What do you think?

**Comment by acdlite** [comment]
The bootstrapping part is definitely nice, and next on my list to implement in Flummox.

But it's using a singleton

https://github.com/goatslacker/iso/blob/master/examples/datetime-flux/src/alt.js

so you're forced to have this awkward separation between user-specific data and regular data. Flummox's approach is far more straightforward, in my opinion. It removes isolation as a concern so you don't have to jump through so many hoops.

I'll start work tonight to get an example up to show you what I mean. Can't make promises on a timeline though, because I'm in the middle of another project. Hopefully by end of week.

**Comment by acdlite** [comment]
I've (rather hastily) thrown together a demo of an isomorphic app using react-router, Immutable.js, and Flummox.

https://github.com/acdlite/flummox-isomorphic-demo

Super basic and needs more attention, but it works. Maybe this will help you out until I find time to work on the docs / make a better demo :)

**Comment by vesparny** [comment]
Hey, thanks for your hints.
They'll be helpful for sure.
I'm quite new to es6 syntax, do you have any good reference to get started about this topic?

Thanks man

**Comment by vesparny** [comment]
@acdlite the process has started :)
I just switched from browserify to webpack in the development branch

**Comment by acdlite** [comment]
6to5 has a nice overview: http://6to5.org/docs/learn-es6/

https://github.com/getify/You-Dont-Know-JS is also a really great resource.

This was actually my first webpack experience. I still <3 browserify, but react-hot-loader alone is worth the switch.

**Comment by vesparny** [comment]
yes, but webpack is also awesome because it supports dynamic require.

[See how many if statements I removed :)](https://github.com/vesparny/morpheus/commit/ca02bf912bfc628a4d216b2e4145bb8a06126066)

**Comment by acdlite** [comment]
Oh wow. Yeah, in my experience gulp + browserify/watchify was never a good combo. browserify works best via the cli. (That was my original motivation for abandoning gulp and just using make, but that's a separate topic, haha.)

**Comment by vesparny** [comment]
I'm trying now to remove gulp watchers in favour of react-hot-loader.
But honestly I don't see the benefit over just rebuild the bundle then reload the page. Am I missing something?

**Comment by acdlite** [comment]
With react-hot-loader, your app maintains its entire state on update. This is awesome if you're working on anything interactive, like animations.

But I agree, for the most part, watchify + livereload (or browsersync) is perfectly fine and almost as fast. Hot loading is just nice to have.

**Comment by vesparny** [comment]
For today the switch to webpack is enough :)

I'll start soon looking at your examples

**Comment by rmahoney-bl** [comment]
very interesting conversation -- thanks guys!
